src/compiler/parser.ts

```diff
@@ -1086,7 +1086,7 @@ namespace ts {
 
         function internIdentifier(text: string): string {
             text = escapeIdentifier(text);
-            return identifiers[text] || (identifiers[text] = text);
+            return getOrUpdate(identifiers, text, () => text);

```

Comment:
This introduces a new closure and environment record for every call to `internIdentifier`. The original approach does not, though this could possibly be worked around by adding `key` to the `makeKey` callback and moving the closure to a separate function declaration.